### ISSUE TITLE
Keyboard switcher disappears after loading xkb config with xkbcomp

### ISSUE THREAD

**Issue by barsv** [awesomeWM/awesome#3688]
Output of `awesome --version`:

awesome v4.3 (Too long)
 • Compiled against Lua 5.3.6 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: 0.9.2

**How to reproduce the issue:**

run:
```
setxkbmap -layout "us,ru" -option "grp:alt_shift_toggle"
# dump current xkb configuration into a tmp file and then reload it back
xkbcomp $DISPLAY tmp.xkb
xkbcomp tmp.xkb $DISPLAY
```
try to switch layout with Shift+Alt.

**Actual result:**

the keyboard layout switcher disappears.
in journalctl i see the message:
```
Sep 05 13:31:33 pop-os /usr/libexec/gdm-x-session[188525]: 2022-09-05 13:31:33 E: awesome: Failed to get list of keyboard groups
```
if i do `return awesome.xkb_get_group_names()` in awesome-client then i get `   string "pc_us_ru_2_inet(evdev)_group(shifts_toggle)_group(alt_shift_toggle)"`
note that it has '\_'(underscore) instead of '+'(plus) as separators between 'us' and 'ru' and '_'(underscore) instead of ':'(colon) between 'ru' and '2'.

**Expected result:**

it should not fail.

**Reason:**

the reason it fails is because the code here -> https://github.com/awesomeWM/awesome/blob/c8d4f878a012bc6d6d0e8e520d17a1dd9237336d/lib/awful/widget/keyboardlayout.lua#L205 expects that xkb_symbols name contains symbol file names separated by "+". this is true if keyboard layout is configured using setxkbmap program. but in general the name of the xkb_symbols section is optional and can be any string.

i haven't looked into xkbcomp code, but AFAIU from observations, to generate a name for xkb_symbol section in binary format xkbcomp either uses the section name of the xkb_symbol section or if there is no section name then it uses the string in the include instruction. when it uses the string in the include then it takes it as is (with pluses). but if it uses the section name then it replaces '+' with '_'. (interesting - why?)

i played with it a bit. for example, i did `xkbcomp -xkm tmp.xkb tmp.xkm` to convert text config into binary config and i see in the generated binary file that the name of the section gets changed - pluses are replaced with underscores. i manually changed underscores to pluses in the binary file and loaded back to xkb with xkbcomp. and it partially worked. the keyboard switcher in awesomewm started to work and layouts were getting changed but for some reason the Ctrl key stopped to work.

**Workaround:**

my actual goal was to add an overlay to the keyboard to use 'hjkl' as arrows in all programs.
the workaround i use is that instead of dumping current layout with xkbcomp (`xkbcomp $DISPLAY my.xkb`) i use setxkbmap (`setxkbmap -print > my.xkb`). it generates a much shorter file with includes instead of a full dump. i add my overalay settings there. 
it's also possible to manually modify files in `/usr/share/X11/xkb/symbols` and then use setxkbmap but i don't like it because i would like to keep my configurations in dotfiles under git.

**Is there a better solution?**

the code in lib/awful/widget/keyboardlayout.lua looks like a hack. it relies on that there are symbol file names in the xkb_symbol section name and the names are separated by "+" while they don't have to be. it doesn't feel good. i was just thinking if there is a better approach. the only idea that i have is that groups in xkb have names. i think that it's possible to get these names directly from xkb and then do to_lowercase and take first 2 chars. with this approach it would be possible to remove all that long list of country codes https://github.com/awesomeWM/awesome/blob/c8d4f878a012bc6d6d0e8e520d17a1dd9237336d/lib/awful/widget/keyboardlayout.lua#L24 and also remove that hacky regex parsing of xkb_symbol name. but group names are again optional and it might now work always. though the workaround can be that in case if group names are not present then just use 'g1', 'g2', etc. it would be better than just exiting with an error as it currently does.

**bottom line**

i resolved my issue and decided to write here about it so that if there will be others who search for it in awesomewm issues they will find it and maybe it will save them time.

the issue doesn't seem to be critical. there is a workarournd. i think that it can be closed for now until better times when we will be getting paid for doing opensource work.
